tests/regression/test_nullspace.py

```diff
@@ -119,6 +119,73 @@ def test_nullspace_mixed():
     assert sqrt(assemble((u - exact)*(u - exact)*dx)) < 5e-8
 
 
+def test_near_nullspace(tmpdir):
+    # Tests the near nullspace for the case of the linear elasticity equations
+    mesh = UnitSquareMesh(100, 100)
+    dim = 2
+    parameters["pyop2_options"]["block_sparsity"] = False
+    V = VectorFunctionSpace(mesh, "Lagrange", 1)
+    u = TrialFunction(V)
+    v = TestFunction(V)
+
+    mu = Constant(0.2)
+    lmbda = Constant(0.3)
+
+    def sigma(fn):
+        return 2.0 * mu * sym(grad(fn)) + lmbda * tr(sym(grad(fn))) * Identity(dim)
+
+    w_exact = Function(V)
+    w_exact.interpolate(Expression(("x[0]*x[1]", "x[0]*x[1]")))
+    # div(sigma(w_exact)) = (mu + lmbda, mu + lmbda)
+    f = Constant((mu + lmbda, mu + lmbda))
+    F = inner(sigma(u), grad(v))*dx + inner(f, v)*dx
+
+    bcs = [DirichletBC(V, w_exact, (1, 2, 3, 4))]
+
+    n0 = Constant((1, 0))
+    n1 = Constant((0, 1))
+    n2 = Expression(("x[1] - 0.5", "-(x[0]-0.5)"))
+    ns = [n0, n1, n2]
+    n_interp = [interpolate(n, V) for n in ns]
+    n_normalized = [interpolate(n*(1.0/sqrt(n.dat.inner(n.dat))), V) for n in n_interp]
+    nsp = VectorSpaceBasis(vecs=n_normalized)
+
+    wo_nns_log = str(tmpdir.join("wo_nns_log"))
+    w_nns_log = str(tmpdir.join("w_nns_log"))
+
+    w1 = Function(V)
+    solve(lhs(F) == rhs(F), w1, bcs=bcs, solver_parameters={
+        'ksp_monitor_short': "ascii:%s:" % w_nns_log,
+        'ksp_rtol': 1e-8, 'ksp_atol': 1e-8, 'ksp_type': 'cg',
+        'pc_type': 'gamg'}, near_nullspace=nsp)
+
+    w2 = Function(V)
+    solve(lhs(F) == rhs(F), w2, bcs=bcs, solver_parameters={
+        'ksp_monitor_short': "ascii:%s:" % wo_nns_log,
+        'ksp_rtol': 1e-8, 'ksp_atol': 1e-8, 'ksp_type': 'cg',
+        'pc_type': 'gamg'})
+
+    # check that both solutions are equal to the exact solution
+    assert sqrt(assemble(inner(w1-w2, w1-w2)*dx)) < 1e-7
+    assert sqrt(assemble(inner(w1-w_exact, w1-w_exact)*dx)) < 1e-7
+
+    with open(wo_nns_log, "r") as f:
+        f.readline()
+        wo = f.read()
+
+    with open(w_nns_log, "r") as f:
+        f.readline()
+        w = f.read()
+
+    print "Without near nullspace"
+    print wo
+    print "With near nullspace"
+    print w
```

Comment:
May I ask why not? If I understand pytest correctly output of successful tests isn't shown anyway.